torch/csrc/jit/export.cpp

```diff
@@ -22,227 +22,176 @@ namespace {
+  RawDataExportMap get_raw_data_export_map() {
+    return raw_data_export_map_;
   }
-  onnx::TensorProto_DataType onnx_type;
-  // Most integral types and float16 need to be serialized as int32
-  at::ScalarType cast_type = tensor.type().scalarType();
-  switch(tensor.type().scalarType()) {
+
+ protected:
+  void EncodeGraph(onnx::GraphProto *graph_proto,
+                   const std::shared_ptr<Graph> &graph,
+                   const std::vector<at::Tensor> &initializers = {});
+
+  void EncodeBlock(onnx::GraphProto *graph_proto,
+                   const Block *block,
+                   const std::vector<at::Tensor> &initializers = {});
+
+  virtual void EncodeTensor(onnx::TensorProto *tensor_proto,
+                            const at::Tensor &tensor,
+                            const at::optional<std::string> external_ref = {});
+
+  virtual void EncodeIntermediateValueInfo(onnx::GraphProto *graph_proto,
+                                           const Value* n) {};
+
+  virtual void EncodeValueInfo(onnx::GraphProto *graph_proto,
+                               onnx::ValueInfoProto* v,
+                               const Value* n);
+
+  void AddAttribute(onnx::NodeProto *node_proto, const jit::Node *node, const jit::Symbol name);
+
+  size_t num_blocks_;
+  bool defer_weight_export_;
+  onnx_torch::OperatorExportTypes operator_export_type_;
+  RawDataExportMap raw_data_export_map_;
+};
+
+onnx::TensorProto_DataType ATenTypeToOnnxType(at::ScalarType at_type) {
+  switch(at_type) {
     case at::kDouble:
-      onnx_type = onnx::TensorProto_DataType_DOUBLE;
-      break;
+      return onnx::TensorProto_DataType_DOUBLE;
     case at::kFloat:
-      onnx_type = onnx::TensorProto_DataType_FLOAT;
-      break;
+      return onnx::TensorProto_DataType_FLOAT;
     case at::kHalf:
-      onnx_type = onnx::TensorProto_DataType_FLOAT16;
-      cast_type = at::kInt;
-      break;
+      return onnx::TensorProto_DataType_FLOAT16;
     case at::kByte:
-      onnx_type = onnx::TensorProto_DataType_UINT8;
-      cast_type = at::kInt;
-      break;
+      return onnx::TensorProto_DataType_UINT8;
     case at::kChar:
-      onnx_type = onnx::TensorProto_DataType_INT8;
-      cast_type = at::kInt;
-      break;
+      return onnx::TensorProto_DataType_INT8;
     case at::kShort:
-      onnx_type = onnx::TensorProto_DataType_INT16;
-      cast_type = at::kInt;
-      break;
+      return onnx::TensorProto_DataType_INT16;
     case at::kInt:
-      onnx_type = onnx::TensorProto_DataType_INT32;
-      break;
+      return onnx::TensorProto_DataType_INT32;
     case at::kLong:
-      onnx_type = onnx::TensorProto_DataType_INT64;
-      break;
+      return onnx::TensorProto_DataType_INT64;
     default:
       AT_ERROR("unexpected tensor scalar type");
-      break;
-  }
-  p->set_data_type(onnx_type);
-  // CPU's HalfTensor doesn't have contiguous(), so first calling contiguous()
-  auto t = tensor.contiguous().toBackend(at::kCPU).toType(cast_type);
```

Comment:
Why did you remove the type casting? ONNX export for integral types != int32 is now broken